### PR TITLE
feat: build wasm using the yomo command

### PR DESCRIPTION
**Pull request by venjiang**
# Description

Build wasm easily and quickly using the yomo command.

## Impact

`yomo run` command adds the `rx` flag, It is used to initialize the SFN of the RxStream `Handler`, the defaults use raw bytes `Handler`.

`yomo build` command adds the `target` flag, It is used to build target, eg: `yomo build --target wasm`

## Guides
1. Initialize the sfn
```sh
yomo init {app-name}
cd {app-name}
```
Write your business processing code in `app.go`.

2. Build 
Requires [tinygo](https://tinygo.org) installation in order to build `wasm`.
```sh
yomo build --target wasm app.go
```

3. Run

```sh
yomo dev sfn.wasm // for development, It's connect to DevZipper(dev.yomo.run:9140)
yomo run sfn.wasm // for production
```

## Example
For details, please visit `example/10-wacli`

### CODE REVIEW

**Review by fanweixiao — APPROVED** [review]
💯

**Comment by fanweixiao** [comment]
I'd like suggest a change for `yomo init`, the default generated code is `rx`, shall we change to `raw`?

`yomo init --rx {app-name}` can generate the rx template.

**Comment by venjiang** [comment]
> I'd like suggest a change for `yomo init`, the default generated code is `rx`, shall we change to `raw`?
> 
> `yomo init --rx {app-name}` can generate the rx template.

Ok, no problem.

File: cli/serverless/golang/serverless.go [hunk 1]
```diff
@@ -191,22 +205,33 @@ func (s *GolangServerless) Build(clean bool) error {
 	// build
 	goos := runtime.GOOS
 	dir, _ := filepath.Split(s.opts.Filename)
-	sl, _ := filepath.Abs(dir + "sl.yomo")
+	sl, _ := filepath.Abs(dir + "sfn.yomo")
```
Comment (fanweixiao) [inline cli/serverless/golang/serverless.go:208]: @venjiang change from `sl.*` to `sfn.*`

Comment (venjiang) [reply cli/serverless/golang/serverless.go:208]: 👌

**Comment by fanweixiao** [comment]
I made some small improvement, and tested this branch on my mac with these steps:

```bash
$ yomo init --raw ccc
⌛  Initializing the Stream Function...
✅  Congratulations! You have initialized the stream function successfully.
ℹ️   You can enjoy the YoMo Stream Function via the command: 
ℹ️       DEV:    cd ccc && yomo dev
ℹ️       PROD:   First run source application, eg: go run example/source/main.go
                Second: cd ccc && yomo run

$ cd ccc

$ tee go.mod <<EOF
module sfn
go 1.19
require (
	github.com/yomorun/yomo v1.12.0-alpha1
)
EOF

$ yomo build --target wasm app.go -m go.mod
ℹ️   YoMo Stream Function file: app.go
⌛  YoMo Stream Function building...
⚠️⚠  Use custom go.mod: /Users/fanweixiao/tmp/ccc/go.mod
✅  Success! YoMo Stream Function build.

$ yomo run -z dev.yomo.run:9140 -n yomo-app-demo sfn.wasm
ℹ️   YoMo Stream Function file: sfn.wasm
⌛  Create YoMo Stream Function instance...
ℹ️   Starting YoMo Stream Function instance with executable file: sfn.wasm. Zipper: [dev.yomo.run:9140].
ℹ️   YoMo Stream Function is running...
time=2023-04-13T10:34:11.471+08:00 level=INFO msg="use credential" component="Stream Function" client_id=WYKGDMLJSoPqzwfcgjj5Y client_name=yomo-app-demo credential_name=none
time=2023-04-13T10:34:12.657+08:00 level=INFO msg="connected to zipper" component="Stream Function" client_id=WYKGDMLJSoPqzwfcgjj5Y client_name=yomo-app-demo zipper_addr=dev.yomo.run:9140
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
sfn received 59 bytes
^C
```

**Pull request merged**